src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -217,17 +268,29 @@ private function createRememberMeServices(ContainerBuilder $container, string $i
         $rememberMeServices->replaceArgument(0, new IteratorArgument(array_unique($userProviders)));
     }
 
-    private function createRememberMeListener(ContainerBuilder $container, string $id, string $rememberMeServicesId): void
+    private function createTokenProvider(ContainerBuilder $container, string $firewallName, array $config): string
     {
-        $container
-            ->setDefinition('security.listener.remember_me.'.$id, new ChildDefinition('security.listener.remember_me'))
-            ->addTag('kernel.event_subscriber', ['dispatcher' => 'security.event_dispatcher.'.$id])
-            ->replaceArgument(0, new Reference($rememberMeServicesId))
-        ;
+        $tokenProviderId = $config['service'];
+        if ($config['doctrine']['enabled'] ?? false) {
+            if (!class_exists(DoctrineTokenProvider::class)) {
+                throw new InvalidConfigurationException('"remember_me.token_provider.doctrine" cannot be enabled as symfony/doctrine-bridge is not installed.');
```

Comment:
```suggestion
                throw new InvalidConfigurationException('Cannot use the "doctrine" token provider for remember_me because symfony/doctrine-bridge is not installed.');
```